src/medida/histogram.cc

```diff
@@ -203,8 +206,8 @@ double Histogram::Impl::variance() const {
 }
 
 
-stats::Snapshot Histogram::Impl::GetSnapshot() const {
-  return sample_->MakeSnapshot();
+stats::Snapshot Histogram::Impl::GetSnapshot(uint64_t divisor = 1) const {
```

Comment:
This isn't quite right, but close. There are two problems:

  - You're providing the default argument to the `Histogram::Impl::GetSnapshot` class member, not the `Histogram::GetSnapshot` class member. The point of the default is to make _clients_ of `Histogram` not have to pass an argument if they don't want to (eg. when someone is using a Histogram outside of a Timer).
  - You're providing the default argument in the out-of-line member function _definition_, where you should be providing it in the _declaration_ (that is: in the `class { ... }` block). I actually didn't know the thing you're doing here works at all! My first thought was that it would not compile. It turns out it's legal but works in a _very odd way_ that's TU-specific -- the rules are [quite subtle](https://en.cppreference.com/w/cpp/language/default_arguments) and I had no idea this special case existed, but .. let's not use it.

In other words, we're going to want:

  - `Histogram::GetSnapshot(uint64_t divisor = 1)` declared inside the `class Histogram` declaration
  - No other default arguments mentioned anywhere else -- not in the `Impl` class and not in any of the out-of-line method definitions. Just that one declaration site.

Reply:
(I made the changes in the next comment instead of this)